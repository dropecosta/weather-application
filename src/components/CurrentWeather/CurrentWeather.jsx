import React, { useState, useContext } from 'react';
import ToggleSwitch from '../ToggleSwitch/ToggleSwitch';
import { AppContext } from '../../context/context';
import './CurrentWeather.scss';

  const CurrentWeather = ({ onChange }) => {
  const { currentCelciusWeather, currentFarenheithWeather } = useContext(AppContext);
  const [temp, setTemp] = useState(Math.round(currentCelciusWeather?.main.temp));
  const [feelsLike, setFeelsLike] = useState(Math.round(currentCelciusWeather?.main.feels_like));
  const [unit, setUnit] = useState("°C");
  const [toggled, setToggled] = useState(false);


  const convert = () => {
    if (toggled) {
      setUnit("°C");
      setTemp(Math.round(currentCelciusWeather?.main.temp));
      setFeelsLike(Math.round(currentCelciusWeather?.main.feels_like));
    } else {
      setUnit("°F");
      setTemp(Math.round(currentFarenheithWeather?.main.temp));
      setFeelsLike(Math.round(currentFarenheithWeather?.main.feels_like));
    }
  };

  const handleChange = (toggled) => {
    onChange(toggled);
}

  return (
    <div className="weather">
      <div className="toggle-container">
        <span className="on">°F</span>
        <ToggleSwitch
          isOn={toggled}
          onColor="#666"
          handleToggle={() => {
            setToggled(!toggled);
            convert();
            handleChange(!toggled)
          }}
        />
        <span className="off">°C</span>
      </div>
      <div className="top">
        <div>
          <p className="city">{currentCelciusWeather?.city}</p>
          <p className="weather-description">{currentCelciusWeather?.weather[0].description}</p>
        </div>
        <img
          alt="weather"
          className="weather-icon"
          src={`icons/${currentCelciusWeather?.weather[0].icon}.png`}
        />
      </div>
      <div className="bottom">
        <p className="temperature">
          {temp}
          {unit}
        </p>
        <div className="details">
          <div className="parameter-row">
            <span className="parameter-label">Feels like</span>
            <span className="parameter-value">
              {feelsLike}
              {unit}
            </span>
          </div>
          <div className="parameter-row">
            <span className="parameter-label">Wind</span>
            <span className="parameter-value">{currentCelciusWeather?.wind.speed} m/s</span>
          </div>
          <div className="parameter-row">
            <span className="parameter-label">Humidity</span>
            <span className="parameter-value">{currentCelciusWeather?.main.humidity}%</span>
          </div>
          <div className="parameter-row">
            <span className="parameter-label">Pressure</span>
            <span className="parameter-value">{currentCelciusWeather?.main.pressure} hPa</span>
          </div>
        </div>
      </div>
    </div>
  );
};

export default CurrentWeather
