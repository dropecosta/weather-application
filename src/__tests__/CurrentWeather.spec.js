import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import CurrentWeather from "../components/CurrentWeather/CurrentWeather";

describe("CurrentWeather component tests", () => {
  it("should render APP component with container class", () => {
    const { container } = render(<CurrentWeather />);
    expect(container.firstChild).toHaveClass("weather");
  });

  it("should match the snapshot", () => {
    const tree = renderer.create(<CurrentWeather />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render text content", () => {
    const { container, getByText } = render(<CurrentWeather />);
    expect(getByText("Feels like")).toBeInTheDocument();
    expect(getByText("Wind")).toBeInTheDocument();
    expect(getByText("Humidity")).toBeInTheDocument();
  });
});
