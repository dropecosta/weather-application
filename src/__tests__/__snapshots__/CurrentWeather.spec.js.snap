// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeather component tests should match the snapshot 1`] = `
<div
  className="weather"
>
  <div
    className="toggle-container"
  >
    <span
      className="on"
    >
      째F
    </span>
    <input
      checked={false}
      className="switch-checkbox"
      id="switch-new"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="switch-label"
      htmlFor="switch-new"
      style={
        Object {
          "background": false,
        }
      }
    >
      <span
        className="switch-button"
      />
    </label>
    <span
      className="off"
    >
      째C
    </span>
  </div>
  <div
    className="top"
  >
    <div>
      <p
        className="city"
      />
      <p
        className="weather-description"
      />
    </div>
    <img
      alt="weather"
      className="weather-icon"
      src="icons/undefined.png"
    />
  </div>
  <div
    className="bottom"
  >
    <p
      className="temperature"
    >
      NaN
      째C
    </p>
    <div
      className="details"
    >
      <div
        className="parameter-row"
      >
        <span
          className="parameter-label"
        >
          Feels like
        </span>
        <span
          className="parameter-value"
        >
          NaN
          째C
        </span>
      </div>
      <div
        className="parameter-row"
      >
        <span
          className="parameter-label"
        >
          Wind
        </span>
        <span
          className="parameter-value"
        >
           m/s
        </span>
      </div>
      <div
        className="parameter-row"
      >
        <span
          className="parameter-label"
        >
          Humidity
        </span>
        <span
          className="parameter-value"
        >
          %
        </span>
      </div>
      <div
        className="parameter-row"
      >
        <span
          className="parameter-label"
        >
          Pressure
        </span>
        <span
          className="parameter-value"
        >
           hPa
        </span>
      </div>
    </div>
  </div>
</div>
`;
